noether_gui/src/widgets/tpp_widget.cpp

```diff
@@ -201,48 +326,71 @@ void TPPWidget::onPlan(const bool /*checked*/)
       throw std::runtime_error("No mesh file selected");
 
     // Load the mesh
-    pcl::PolygonMesh mesh;
-    if (pcl::io::loadPolygonFile(mesh_file, mesh) < 1)
+    pcl::PolygonMesh full_mesh;
+    if (pcl::io::loadPolygonFile(mesh_file, full_mesh) < 1)
       throw std::runtime_error("Failed to load mesh from file");
 
     const ToolPathPlannerPipeline pipeline = pipeline_widget_->createPipeline();
     QApplication::setOverrideCursor(Qt::WaitCursor);
-    tool_paths_ = pipeline.plan(mesh);
-    QApplication::restoreOverrideCursor();
 
-    // Render the tool paths
-    std::for_each(tool_path_actors_.begin(), tool_path_actors_.end(), [this](vtkProp* actor) {
-      renderer_->RemoveActor(actor);
-      actor->Delete();
-    });
-    tool_path_actors_.clear();
+    // Run the mesh modifier
+    std::vector<pcl::PolygonMesh> meshes = pipeline.mesh_modifier->modify(full_mesh);
+
+    tool_paths_.clear();
+    tool_paths_.reserve(meshes.size());
 
-    for (const ToolPaths& fragment : tool_paths_)
+    pcl::PolygonMesh combined_mesh_fragments;
+    std::vector<ToolPaths> unmodified_tool_paths;
+    std::string file_name = "/tmp/comb_frag_mesh_test";
+    for(std::size_t i = 0; i < meshes.size(); ++i)
     {
-      for (const ToolPath& tool_path : fragment)
+      const pcl::PolygonMesh& mesh = meshes[i];
+      if (i == 0)
       {
-        for (const ToolPathSegment& segment : tool_path)
-        {
-          for (const Eigen::Isometry3d& w : segment)
-          {
-            auto transform_filter = vtkSmartPointer<vtkTransformFilter>::New();
-            transform_filter->SetTransform(toVTK(w));
-            transform_filter->SetInputConnection(tube_filter_->GetOutputPort());
+        combined_mesh_fragments = mesh;
+      }
 
-            auto map = vtkSmartPointer<vtkPolyDataMapper>::New();
-            map->SetInputConnection(transform_filter->GetOutputPort());
+      else
+      {
+        combined_mesh_fragments = combineMeshes(combined_mesh_fragments, mesh);
+      }
```

Comment:
Is the if-statement necessary? You may be able to call `combineMeshes` when the first mesh is empty

Reply:
Not necessary with call to updated `concatenate` function